sdk/cosmos/azure-cosmos-spark_3/src/main/scala/com/azure/cosmos/spark/CosmosItemsDataSource.scala

```diff
@@ -112,4 +112,182 @@ object CosmosItemsDataSource {
 
     readManyReader.readMany(df.rdd, readManyFilterExtraction)
   }
+
+  /**
+   * Execute transactional batch operations against Cosmos DB.
+   * Operations with the same partition key are executed atomically.
+   * 
+   * Input DataFrame schema:
+   * - Flat columns corresponding to the properties of the Cosmos DB item to be operated on.
+   * - The partition key column name must match the container's partition key path (e.g., "pk" if the path is "/pk").
+   * - The "id" column (String) is required.
+   * - An optional "operationType" column (String) can be provided to specify the operation ("create", "replace", "upsert", "delete", "read") for each row.
```

Comment:
transactional read should not be supporte din Spark - just a way to shoot yourself in the foot